sympy/core/basic.py

```diff
@@ -1132,6 +1132,30 @@ def has(self, *patterns):
         >>> x.has(x)
         True
 
+        Note ``has`` is a structural algorithm with no knowledge of
+        mathematics. Assuming otherwise can lead to what seem to be
```

Comment:
I would keep it brief with one good example. Maybe just the Interval example, showing the interval so one can see that it is open at the end points:

```
...mathematics. Consider the following interval:

>>> from sympy.sets import Interval
>>> i = Interval.Lopen(0, 5); i
(0, 5]
>>> i.has(4)  # there is no "4" in the arguments
False
>>> i.has(0)  # there *is* a "0" in the arguments
True

The use of `contains` will tell whether a number is in the interval or not, however:

>>> i.contains(4)
True
>>> i.contains(0)
False
```